octoprint_mrbeam/__init__.py

```diff
@@ -1296,6 +1300,33 @@ def generate_backlash_compenation_pattern_gcode(self, data):
             res = dict(calibration_pattern=destFile, target=FileDestinations.LOCAL)
             return jsonify(res)
 
+    # simpleApiCommand: compare_pep440_versions;
+    def compare_pep440_versions(self, data):
+        """
+            returns the PEP440 version comparison Boolean result
+
+            Args:
+                data (dict): Includes the following keys:
+                    v1 (str): First version to be compared
+                    v2 (str): Second version to be compared
+                    operator (str): Comparison operator
+
+            Returns:
+                Boolean: PEP440 version comparison result
+        """
+        self._logger.info("compare_pep440_versions() request: %s", data)
+        try:
+            v1 = pkg_resources.parse_version(data.get("v1", None))
+            v2 = pkg_resources.parse_version(data.get("v2", None))
+            comparator = data.get("operator", None)
+            result = VersionComparator.get_comparator(
+                comparator, COMPARISON_OPTIONS
+            ).compare(v1, v2)
+            return make_response(json.dumps(result), 200)
+        except Exception as e:
```

Comment:
Please handle the exception or catch before if one of the parameters is missing

Reply:
Done